### PR TITLE
8305169: java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java -- test server didn't start in timely manner

### PR DESCRIPTION
**Pull request by mpdonova**
Increased the time to wait for the server thread to start and if it doesn't; we stop and restart it. After a minute, if the server hasn't started, throw an exception.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305169](https://bugs.openjdk.org/browse/JDK-8305169): java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java -- test server didn't start in timely manner

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13559/head:pull/13559` \
`$ git checkout pull/13559`

Update a local copy of the PR: \
`$ git checkout pull/13559` \
`$ git pull https://git.openjdk.org/jdk.git pull/13559/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13559`

View PR using the GUI difftool: \
`$ git pr show -t 13559`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13559.diff">https://git.openjdk.org/jdk/pull/13559.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java [hunk 1]
```diff
@@ -169,6 +161,20 @@ public static void main(String args[]) throws Exception {
         }
     }
 
+    private static void tryToStartOcspServer(SimpleOCSPServer ocspResponder) throws InterruptedException, IOException {
```
Comment (msheppar) [inline test/jdk/java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java:164]: startOcspServer would be fine

File: test/jdk/java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java [hunk 2]
```diff
@@ -169,6 +161,20 @@ public static void main(String args[]) throws Exception {
...
+        for (int i = 0 ; i < 6 ; ++i) {
```
Comment (msheppar) [inline test/jdk/java/security/cert/CertPathValidator/OCSP/GetAndPostTests.java:165]: replace the iterations upper bound with a constant

private static final int MAX_SERVER_START_RETRY = 6;

or whatever you prefer as its moniker

File: test/jdk/java/security/testlibrary/SimpleOCSPServer.java [hunk 3]
```diff
@@ -292,6 +292,13 @@ public synchronized void stop() {
         }
     }
 
+    public synchronized void shutdownNow() {
```
Comment (msheppar) [inline test/jdk/java/security/testlibrary/SimpleOCSPServer.java:295]: This is the tricky one how to make shutdownNow precise and effective. It is not clear for the jtreg timeout where the server start has stalled in the launch of the service thread. Is it stuck in the the executors thread pool, or has the thread been created an is on the Ready to Run queue of the OS due to heavy load on system. The server thread doesn't handle a Thread.interrupt, and it inferred that this mechanism used by the ExecutorService to terminate a task then  the try within the run method would need to handle InterruptedException and invoke some shutdown e.g. close the server socket etc.  Or will that be taken care by try with resources and autoclave. We known or can reasonably assume that the server thread has not invoked the count latch countdown method.  One could theorise that if the server thread got bumped after the bind but during the log call (as its IO) then there will be potential be some server socket left open.  I think it would be worthwhile to emulate where the server thread doesn't start in timely manner by inserting a Thread.sleep in the run method sleep > 10 seconds and see if the restart works effectively and that multiple server sockets are not left open
In short need to look at the shotdown processing more closely to ensure that it does the necessary and appropriate cleanup for each retry iteration
